core/src/test/java/org/sonarsource/sonarlint/core/container/storage/FileMatcherTest.java

```diff
@@ -103,6 +100,46 @@ public void should_return_most_common_prefixes() {
     assertThat(match.sqPrefix()).isEqualTo(Paths.get("sq1"));
   }
 
+  @Test
+  public void should_favor_deepest_common_path() {
+    List<Path> idePaths = Arrays.asList(
+      Paths.get("local1/pom.xml"),
+      Paths.get("local1/build.properties"),
+      Paths.get("local1/src/main/java/com/foo/A.java"));
+
+    List<Path> sqPaths = Arrays.asList(
+      Paths.get("sq1/pom.xml"),
+      Paths.get("sq1/build.properties"),
+      Paths.get("sq2/src/main/java/com/foo/A.java")
+
+    );
+    FileMatcher.Result match = fileMatcher.match(sqPaths, idePaths);
+    assertThat(match.idePrefix()).isEqualTo(Paths.get("local1"));
+    assertThat(match.sqPrefix()).isEqualTo(Paths.get("sq2"));
+  }
+
+  @Test
+  public void should_disfavor_path_having_multiple_matches() {
+    List<Path> idePaths = Arrays.asList(
+      Paths.get("local1/pom.xml"),
+      Paths.get("local1/build.properties"),
+      Paths.get("local1/src/A.java"));
+
+    List<Path> sqPaths = Arrays.asList(
+      Paths.get("sq1/pom.xml"),
+      Paths.get("sq1/build.properties"),
+      Paths.get("sq2/pom.xml"),
+      Paths.get("sq2/build.properties"),
+      Paths.get("sq3/pom.xml"),
+      Paths.get("sq3/build.properties"),
+      Paths.get("sq4/src/A.java")
```

Comment:
With your proposed change, we are back at testing that the prefix with the biggest number of file matched is the preferred one, which was already the case before. Here I want to test that sq4 will be selected even if this is not the one having the most matched files, but because `pom.xml` and `build.properties` are matched many times, their score will be lower than the single match of `src/A.java`